### ISSUE TITLE
Add a Shared Instance of Ajv

### ISSUE THREAD

**Issue by MitchellMcKenna** [a85/POSTMan-Chrome-Extension#6590]
**Describe the problem**
I read tv4 is deprecated in favor of Ajv [[1](https://github.com/postmanlabs/postman-app-support/issues/3173#issuecomment-492345566)][[2](https://github.com/postmanlabs/postman-app-support/issues/3199)], however the way you access Ajv is different than how you access tv4. With Ajv you have to require the package and new up a new instance, whereas with tv4 it is available as a shared instance across collection/folder pre-request scripts as well as the requests tests. 

I find the shared instance particularly valuable when using the Runner/Newman to fetch/load the schema one time inside the collection pre-request script and use it over and over again across all my requests/tests, which is much faster than fetching the schemas over and over again inside each request's tests. Here's an example of how I'm able to have schemas only loaded once via the collection pre-request script:
```js
if (Object.keys(tv4.getSchemaMap()).length === 0) {
    const openapiUrl = 'http://example.com/openapi.json';

    pm.sendRequest(openapiUrl, function (err, res) {
        tv4.addSchema(openapiUrl, res.json())
    });
}
```

**Describe the solution you'd like**
Please add a shared instance of ajv to be used between requests.

Additional benefits:
- Makes it easier to migrate users over from tv4 to Ajv because it makes accessing it similar.
- No need to require the package and new up a new instance in every single test script.
- Since Ajv is already instantiated there wouldn't be the issue of users forgetting to pass `{logger: console}` in the constructor and wondering why Ajv doesn't work for them. #6400
- Users who like to new up a new instance of Ajv could continue to do so.

**Comment by shamasis** [comment]
The shared instance concept is something that we deprecated since it was getting in the way of visibility and conflicts with other scripts. It was also causing debugging issues. Having said that, the I understand the pain of logger issue and we are working to wrap AJV constructor to pre-inject console. 

Additionally, two scripts in sandbox are isolated from each other, hence even though the older module variable (_, etc) appears global, but they are separate instances. Hence, this would be tricky. For now, it seems that adding schema in top level scripts and pass the instance to other scripts, is the only viable features.

**Comment by MitchellMcKenna** [comment]
@shamasis How would one "pass the instance to other scripts" as you suggest?

**Comment by shamasis** [comment]
Yes. Exactly. That’s the problem. The way we can achieve that is break script execution isolation while running scripts from collection level and then figure out a way to pass functions using “some method”. Both needs to be built. I’m just throwing ideas and possibilities. What comes out finally would be what fits best and adds most value. So, more feedback here would be great.

**Comment by appurva21** [comment]
Hi @MitchellMcKenna! 👋

You can achieve a shared instance across all folders and requests in a collection by requiring and instantiating Ajv in a Collection-level scripts (Pre-request or Post-response) as a global variable. 
Here's how:

1. Add this to your collection’s pre-request script:
```js
const Ajv = pm.require("npm:ajv");

// delcare it as global
ajv = new Ajv({ strict: false });

ajv.addSchema({
    $id: "http://example.com/openapi.json",
    type: "object",
    properties: {
        foo: { type: "string" }
    },
    required: ["foo"]
});
``` 

2. Now you can use the `ajv` variable across any folder/request scripts within that collection:
```js
// Reference the schema ID used while adding the schema
const validate = ajv.getSchema("http://example.com/openapi.json");

const valid = validate({ foo: "bar" });
console.log("Is valid?", valid);
```

📦 Note: Postman now supports importing packages from NPM directly into your scripts. We recommend using this going forward instead of the older way of requiring Ajv:
```js
// Recommended
const Ajv = pm.require("npm:ajv");

// Old syntax
const Ajv = require("ajv");
```

For more on how to use external packages in scripts, check out: https://github.com/postmanlabs/postman-app-support/issues/13542

Hope this helps!
